src/UrlModal.tsx

```diff
@@ -0,0 +1,150 @@
+import React, { useState, useEffect, Suspense, useCallback } from 'react';
+import { MODAL_KEY, PARAMS_KEY } from './constants';
+import { createURL, decodedUrlParams, encodeUrlParams } from './helpers';
+import useCustomEvent from './hooks/useCustomEvent';
+
+const routerPush = async (href: string) =>
+  window.history.pushState({ path: href }, '', href);
+
+export const openModal = ({
```

Comment:
Why not export these helpers too? Just curious, they are fine either way to me.

Reply:
Truu, should export the open modal

Done

Reply:
Wait i did bad things